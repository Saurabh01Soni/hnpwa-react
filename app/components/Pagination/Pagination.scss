$_pagination-box-shadow-color: rgba(0, 0, 0, .1);

.Pagination {
  box-shadow: 0 1px 2px $_pagination-box-shadow-color;
  height: $header-height;
  text-align: center;
  width: 100%;
}

.Pagination__inner {
  display: inline-block;
  padding: 18px 0;
}

.Pagination__button {
  background-color: transparent;
  border: 0;
}

.Pagination__button--disabled {
  color: $color-gray;
}

.Pagination__number {
  display: inline-block;
  margin: 0 20px;
}
