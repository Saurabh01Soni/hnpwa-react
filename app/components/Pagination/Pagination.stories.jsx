import React from 'react';
import { storiesOf } from '@storybook/react';

import Pagination from './';

const stories = storiesOf('Pagination', module);

class PaginationWrapper extends React.Component {
  constructor(props) {
    super(props);
    this.state = { currentPage: 1, maxPage: 10 };
    this.handlePaginate = this.handlePaginate.bind(this);
  }

  handlePaginate(number) {
    this.setState({ currentPage: number });
  }

  render() {
    return (
      <Pagination
        currentPage={this.state.currentPage}
        maxPage={this.state.maxPage}
        onPaginate={this.handlePaginate}
      />
    );
  }
}

stories
  .add('Pagination', () => <PaginationWrapper />);
