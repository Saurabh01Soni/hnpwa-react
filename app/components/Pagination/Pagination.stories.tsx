import * as React from 'react';
import { storiesOf } from '@storybook/react';

import Pagination from './';

const stories = storiesOf('Pagination', module);

interface IPaginationWrapperState {
  currentPage: number;
}

class PaginationWrapper extends React.Component<{}, IPaginationWrapperState> {
  constructor(props) {
    super(props);
    this.state = { currentPage: 1 };
    this.handlePaginate = this.handlePaginate.bind(this);
  }

  public handlePaginate(value: number) {
    this.setState({ currentPage: value });
  }

  public render() {
    return (
      <Pagination
        currentPage={this.state.currentPage}
        onPaginate={this.handlePaginate}
      />
    );
  }
}

stories
  .add('Pagination', () => <PaginationWrapper />);
