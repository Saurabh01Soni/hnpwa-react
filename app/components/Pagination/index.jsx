import cx from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';

import './Pagination.scss';

const propTypes = {
  currentPage: PropTypes.number,
  maxPage: PropTypes.number,
  onPaginate: PropTypes.func,
};

const defaultProps = {
  currentPage: 1,
  maxPage: 1,
  onPaginate() {},
};

function Pagination({ currentPage, maxPage, onPaginate }) {
  const prevDisabled = currentPage === 1;
  const nextDisabeld = currentPage === maxPage;

  const prevButtonClassName = cx('Pagination__button', {
    'Pagination__button--disabled': prevDisabled,
  });

  const nextButtonClassName = cx('Pagination__button', {
    'Pagination__button--disabled': nextDisabeld,
  });

  return (
    <div className="Pagination">
      <div className="Pagination__inner">
        <button
          className={prevButtonClassName}
          disabled={prevDisabled}
          type="button"
          onClick={() => onPaginate(currentPage - 1)}
        >
          {'<'} Prev
        </button>
        <div className="Pagination__number">{currentPage} / {maxPage}</div>
        <button
          disabled={nextDisabeld}
          className={nextButtonClassName}
          type="button"
          onClick={() => onPaginate(currentPage + 1)}
        >
          {'>'} Next
        </button>
      </div>
    </div>
  );
}

Pagination.propTypes = propTypes;
Pagination.defaultProps = defaultProps;

export default Pagination;
