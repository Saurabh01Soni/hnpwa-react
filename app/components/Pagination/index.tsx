import * as cx from 'classnames';
import * as React from 'react';

import './Pagination.scss';

export interface IPaginationProps {
  readonly currentPage?: number;
  readonly onPaginate?: (value: number) => void;
}

const defaultProps: IPaginationProps = {
  currentPage: 1,
  onPaginate() {},
};

const Pagination: React.SFC<IPaginationProps> = ({ currentPage, onPaginate }) => {
  const prevDisabled = currentPage === 1;
  const nextDisabled = currentPage === 10;

  const prevButtonClassName = cx('Pagination__button', {
    'Pagination__button--disabled': prevDisabled,
  });

  const nextButtonClassName = cx('Pagination__button', {
    'Pagination__button--disabled': nextDisabled,
  });

  return (
    <div className="Pagination">
      <div className="Pagination__inner">
        <button
          className={prevButtonClassName}
          data-button="prev"
          disabled={prevDisabled}
          type="button"
          onClick={() => onPaginate(currentPage - 1)}
        >
          {'<'} Prev
        </button>
        {currentPage} / 10
        {' '}
        <button
          className={nextButtonClassName}
          data-button="next"
          disabled={nextDisabled}
          type="button"
          onClick={() => onPaginate(currentPage + 1)}
        >
          Next {'>'}
        </button>
      </div>
    </div>
  );
};

Pagination.defaultProps = defaultProps;

export default Pagination;
