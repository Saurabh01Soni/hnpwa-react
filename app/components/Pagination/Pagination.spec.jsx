import React from 'react';
import { shallow } from 'enzyme';

import Pagination from './';

describe('<Pagination />', () => {
  it('should match snapshot when render default', () => {
    const wrapper = shallow(<Pagination />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should calls onPaginate with currentPage - 1 when simulate prev button click event', () => {
    const currentPage = 3;
    const onPaginate = jest.fn();
    const wrapper = shallow(
      <Pagination
        currentPage={currentPage}
        maxPage={10}
        onPaginate={onPaginate}
      />,
    );
    wrapper.find('.Pagination__button').at(0).simulate('click');
    expect(onPaginate.mock.calls.length).toBe(1);
    expect(onPaginate.mock.calls[0]).toEqual([currentPage - 1]);
  });

  it('should calls onPaginate with currentPage + 1 when simulate next button click event', () => {
    const currentPage = 3;
    const onPaginate = jest.fn();
    const wrapper = shallow(
      <Pagination
        currentPage={currentPage}
        maxPage={10}
        onPaginate={onPaginate}
      />,
    );
    wrapper.find('.Pagination__button').at(1).simulate('click');
    expect(onPaginate.mock.calls.length).toBe(1);
    expect(onPaginate.mock.calls[0]).toEqual([currentPage + 1]);
  });
});
