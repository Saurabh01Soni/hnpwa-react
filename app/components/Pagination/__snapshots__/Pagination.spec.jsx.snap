// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should have .Pagination__button--disabled when props.currentPage === props.maxPage 1`] = `
<div
  className="Pagination"
>
  <div
    className="Pagination__inner"
  >
    <button
      className="Pagination__button"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      &lt;
       Prev
    </button>
    <div
      className="Pagination__number"
    >
      10
       / 
      10
    </div>
    <button
      className="Pagination__button Pagination__button--disabled"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      &gt;
       Next
    </button>
  </div>
</div>
`;

exports[`<Pagination /> should match snapshot when render default 1`] = `
<div
  className="Pagination"
>
  <div
    className="Pagination__inner"
  >
    <button
      className="Pagination__button Pagination__button--disabled"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      &lt;
       Prev
    </button>
    <div
      className="Pagination__number"
    >
      1
       / 
      1
    </div>
    <button
      className="Pagination__button Pagination__button--disabled"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      &gt;
       Next
    </button>
  </div>
</div>
`;
